### ISSUE TITLE
Unexpected behaviour with C/C++ extension and typedef

### ISSUE THREAD

**Issue by Jamen1993** [eamodio/vscode-gitlens#425]
- GitLens Version: 8.4.0
- VSCode Version: 1.24.1
- OS Version: W10 1803

I use GitLens together with Microsofts C/C++ extension and it shows file edit information before each typedef. I'm not sure whether this is intended behaviour but I don't think so.

![image](https://user-images.githubusercontent.com/17482299/41606527-26a3abb6-73e4-11e8-8021-6b58b816f422.png)

![image](https://user-images.githubusercontent.com/17482299/41606555-37898b8a-73e4-11e8-8acf-4e4ce04484fa.png)

**Comment by eamodio** [comment]
Can you try setting "gitlens.debug": true and then send a screen shot of those code lens again? That will give the type of symbol of that type and it can be turned off directly.

**Comment by Jamen1993** [comment]
Hi eamodio,

please excuse that it took me so long to collect the requested screens.

![first](https://user-images.githubusercontent.com/17482299/42125017-d3aa4d1e-7c6e-11e8-9c18-26d6bad0652e.png)
![second](https://user-images.githubusercontent.com/17482299/42125018-d7c448a0-7c6e-11e8-88cb-f01d687e9bdc.png)

Is this what you need?

**Comment by eamodio** [comment]
Yes thank you. It seems a bit odd to me that the C++ language server is using `SymbolKind.Class` for all those (rather than `SymbolKind.Enum` and `SymbolKind.Struct`), but regardless the issue would be largely the same. 

While it would affect more than just these `typedef`s you can turn off the code lens on the `container` scope by setting `"gitlens.codeLens.scopes"= ["document"]`.

Or you add to the `gitlens.codeLens.scopesByLanguage` setting to only set the `document` scope for `"language": "cpp"`
```json
{
    "language": "cpp",
    "scopes": [
        "document"
    ]
},
```

Or you add to the `gitlens.codeLens.scopesByLanguage` setting to only unset the `Class` symbol for `"language": "cpp"`
```json
{
    "language": "cpp",
        "symbolScopes": [
            "!Class"
        ]
},
```

Maybe once https://github.com/Microsoft/vscode/issues/34968 lands and the language services provide more information I might be able to do more here to improve this behavior.

**Comment by eamodio** [comment]
Depends on: https://github.com/Microsoft/vscode/issues/57984

**Comment by gajop** [comment]
I'm having a similar issue with `using MyFunctionType = std::function<void()>;`
I get author + date information above each such line.
